doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md

```diff
@@ -0,0 +1,582 @@
+             |     |  +--:(ietf-nat-ext:ip-address)
+             |     |  |  +--rw ietf-nat-ext:IP-ADDRESS?         inet:ipv4-address
+             |     |  +--:(ietf-nat-ext:ip-address-range)
+             |     |     +--rw ietf-nat-ext:IP-ADDRESS-RANGE?   string
+             |     +--rw ietf-nat-ext:nat-port?           string
+             +--rw ietf-nat-ext:nat-bindings
+                +--rw ietf-nat-ext:nat-binding-entry* [binding-name]
+                   +--rw ietf-nat-ext:binding-name    string
+                   +--rw ietf-nat-ext:nat-pool        string
+                   +--rw ietf-nat-ext:access-list?    string
+                   +--rw ietf-nat-ext:nat-type?       enumeration
+                   +--rw ietf-nat-ext:twice-nat-id?   uint16
+
+```
+
+SONiC NAT Yang will be used for Config Validation purposes.
+
+https://github.com/project-arlo/sonic-mgmt-framework/blob/nat-impl/models/yang/sonic/sonic-nat.yang
+
+### 3.6.2 CLI
+SONiC NAT Click CLI based Configuration and Show Commands will be supported by management framework.
+Refer Section 3.8 from https://github.com/Azure/SONiC/blob/dc5d3a894618bcb07a3c5d2dd488caf3beb7479a/doc/nat/nat_design_spec.md
+
+#### 3.6.2.1 Configuration Commands
+
+###### Add static NAT entry
+
+`static basic <global-ip> <local-ip> {snat | dnat} { twice_nat_id <value> }`
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat)# static basic 20.0.0.2 65.55.45.8 snat twice_nat_id 1
+```
+###### Remove static NAT entry
+
+`
+sonic# config t
+sonic(config)# nat
+sonic(config-nat) # no static basic {global-ip} {local-ip}`
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat)# no static basic 20.0.0.2 65.55.45.8
+```
+
+###### Add static NAPT entry
+
+`static {tcp | udp} <global-ip> <global-port> <local-ip> <local-port> {snat | dnat} { twice_nat_id <value> }`
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat)# static tcp 65.55.45.7 4000 20.0.0.1 4500 dnat twice_nat_id 1
+```
+
+###### Remove static NAPT entry
+
+`no static {tcp | udp} <global-ip> <global-port> <local-ip> <local-port>`
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat)# no static tcp 65.55.45.7 4000 20.0.0.1 4500
+```
+
+###### Remove all static NAT/NAPT configuration
+
+`no static all`
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat) # no static all
+```
+
+###### Create NAT pool
+
+`pool <pool-name> <global-ip-range> <global-port-range>`
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat) pool Pool1 65.55.45.10-65.55.45.15 1024-65535
+```
+
+###### Remove NAT pool
+
+`no pool <pool-name>`
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat) no pool Pool1
+```
+
+###### Remove all NAT pool configuration
+
+`no pools`
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat) no pools
+```
+
+###### Create binding between an ACL and a NAT pool
+
+`binding <binding-name> <pool-name> <acl-name> { snat | dnat } { twice_nat_id <value> }`
```

Comment:
{ snat | dnat } and twice_nat_id are optional parameters.

Reply:
Addressed